cmake/libmultiprocess.cmake

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2025 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://opensource.org/license/mit/.
+
+function(add_libmultiprocess subdir)
+  # Unconditionally set BUILD_TESTING to ON. BUILD_TESTING is a standard cmake
+  # variable that controls whether CTest is used. In the bitcoin build, CTest is
+  # always used and we do not want an option to turn it off. But in
+  # libmultiprocess, CTest is optional and BUILD_TESTING needs to be set in
+  # order to add libmultiprocess unit tests to the CTest configuration.
+  set(BUILD_TESTING ON)
+  add_subdirectory(${subdir} EXCLUDE_FROM_ALL)
+  # Apply core_interface compile options to libmultiprocess runtime library.
+  target_link_libraries(multiprocess PUBLIC $<BUILD_INTERFACE:core_interface>)
+  target_link_libraries(mputil PUBLIC $<BUILD_INTERFACE:core_interface>)
+  target_link_libraries(mpgen PUBLIC $<BUILD_INTERFACE:core_interface>)
```

Comment:
88eabc8c2c0fbb62db5cb6248cc8ab770f3ce78b

Why does building the `mpgen` tool require `core_interface`?

Reply:
Having the vintage for each file is nice :-)